src/odin/worker.cc

```diff
@@ -55,30 +49,19 @@ odin_worker_t::work(const std::list<zmq::message_t>& job,
                     const std::function<void()>& interrupt_function) {
   auto& info = *static_cast<prime_server::http_request_info_t*>(request_info);
   LOG_INFO("Got Odin Request " + std::to_string(info.id));
-  valhalla_request_t request;
+  Api request;
   try {
-    // crack open the original request
-    std::string serialized_options(static_cast<const char*>(job.front().data()), job.front().size());
-    request.parse(serialized_options);
-
     // Set the interrupt function
     service_worker_t::set_interrupt(interrupt_function);
 
-    // parse each leg
-    std::list<TripLeg> legs;
-    for (auto leg = ++job.cbegin(); leg != job.cend(); ++leg) {
-      // crack open the path
-      legs.emplace_back();
-      try {
-        legs.back().ParseFromArray(leg->data(), static_cast<int>(leg->size()));
-      } catch (...) { return jsonify_error({201}, info, request); }
-    }
+    // crack open the in progress request
+    request.ParseFromArray(job.front().data(), job.front().size());
```

Comment:
odin and and thor look like this now. they just turn the single message of bytes into the pbf